src/lib.rs

```diff
@@ -645,10 +639,7 @@ pub mod pallet {
 		) -> Result<(), &'static str> {
 			log::info!("🐙 in observing_mainchain");
 
-			let mut next_fact_sequence = 0;
-			if let Some(cur_seq) = FactSequence::<T>::get() {
-				next_fact_sequence = cur_seq + 1;
-			}
+			let next_fact_sequence = NextFactSequence::<T>::get() + 1;
```

Comment:
no need to +1

Reply:
Update in https://github.com/octopus-network/pallet-octopus-appchain/pull/8/commits/6309898a8f0b755c9768623148750a3e9a2f7165